### PR TITLE
Move more video classifiers

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Move more video classifiers to the VideoFileItemClassify source.

## Motivation and context
Reduce CFileItem API.

## How has this been tested?
Is builds and I have added tests.

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/test/TestVideoFileItemClassify.cpp [hunk 1]
```diff
@@ -144,6 +153,13 @@ TEST(TestVideoFileItemClassify, IsSubtitle)
   EXPECT_FALSE(VIDEO::IsSubtitle(CFileItem("random.notasub", false)));
 }
 
+TEST(TestVideoFileItemClassify, IsVideoDb)
+{
+  EXPECT_TRUE(VIDEO::IsVideoDb(CFileItem("videodb://1/2/3", false)));
+  EXPECT_TRUE(VIDEO::IsVideoDb(CFileItem("videodb://1/2/", true)));
+  EXPECT_FALSE(VIDEO::IsVideoDb(CFileItem("/videodb/home/foo/Extraordinary/", true)));
+}
+
```
Comment (CrystalP) [inline xbmc/video/test/TestVideoFileItemClassify.cpp:205]: uh? in the wrong commit I guess

Comment (notspiff) [reply xbmc/video/test/TestVideoFileItemClassify.cpp:205]: indeed. moved.

File: xbmc/video/test/TestVideoFileItemClassify.cpp [hunk 2]
```diff
@@ -143,3 +143,10 @@ TEST(TestVideoFileItemClassify, IsSubtitle)
 
   EXPECT_FALSE(VIDEO::IsSubtitle(CFileItem("random.notasub", false)));
 }
+
+TEST(TestVideoFileItemClassify, IsVideoExtras)
+{
+  EXPECT_TRUE(VIDEO::IsVideoExtras(CFileItem("/home/foo/Extras/", true)));
+  EXPECT_TRUE(VIDEO::IsVideoExtras(CFileItem("/home/foo/extras/", true)));
+  EXPECT_FALSE(VIDEO::IsVideoExtras(CFileItem("/home/foo/Extraordinary/", true)));
```
Comment (CrystalP) [inline xbmc/video/test/TestVideoFileItemClassify.cpp:210]: I'd suggest another test for "/home/foo/Extras/abc.mkv". Expected result false.

File: xbmc/video/VideoFileItemClassify.cpp [hunk 3]
```diff
@@ -83,4 +83,10 @@ bool IsVideo(const CFileItem& item)
                                 CServiceBroker::GetFileExtensionProvider().GetVideoExtensions());
 }
 
+bool IsVideoExtras(const CFileItem& item)
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/video/VideoFileItemClassify.cpp]: i agree. have added a commit addressing this.

File: xbmc/video/test/TestVideoFileItemClassify.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/video/test/TestVideoFileItemClassify.cpp]: added.

**Comment by notspiff** [comment]
Anything else?

**Review by CrystalP — COMMENTED** [review]
LGTM

**Comment by notspiff** [comment]
I'll read dat as an approval.

**Pull request merged**